src/java.base/share/classes/java/util/UUID.java

```diff
@@ -181,6 +181,46 @@ public static UUID nameUUIDFromBytes(byte[] name) {
         return new UUID(md5Bytes);
     }
 
+    /**
+     * Static factory to retrieve a type 7 (time based) {@code UUID} based on
+     * the current timestamp, enhanced with sub-millisecond precision.
+     *
+     * The {@code UUID} embeds the current Unix timestamp in milliseconds into
+     * the first 6 bytes, sets the version and variant bits as per RFC 9562,
+     * replaces the first 12 random bits with a value derived from the current
+     * system clock's sub-millisecond precision, and fills the
+     * remaining bytes with cryptographically strong random data.
+     *
+     * @return A {@code UUID} generated from the current system time
+     *
+     * @spec RFC 9562
+     */
+    public static UUID timestampUUID() {
+        long msTime = System.currentTimeMillis();
+        long nsTime = System.nanoTime();
+        SecureRandom ng = Holder.numberGenerator;
+        byte[] randomBytes = new byte[16];
+        ng.nextBytes(randomBytes);
```

Comment:
It's a bit unfortunate that `SecureRandom#nextBytes(byte[])` does not allow us to pass an offset and length. Here we generate 16 bytes of random data but throw the first 8 bytes away.

Reply:
@kieran-farrell Is there a last-minute plan to make use of `nextLong()` as suggested by @RogerRiggs?
Otherwise looks good

Reply:
This thread can be resolved.